tests/test_views.py

```diff
@@ -3942,6 +3942,23 @@ def test_external_login_confirm_email_get_create(self, mock_welcome):
         assert_true(self.user.is_registered)
         assert_true(self.user.has_usable_password())
 
+    def test_external_login_confirm_email_get_while_logged_in(self):
+        external_identity = {
+            'orcid': {
+                self.provider_id: 'PENDING'
+            }
+        }
+        user = AuthUserFactory(external_identity=external_identity)
```

Comment:
Could you modify `self.user` for your test instead of taking the time to create your own?